smack-bosh/src/main/java/org/jivesoftware/smack/bosh/XMPPBOSHConnection.java

```diff
@@ -537,7 +538,12 @@ public void responseReceived(BOSHMessageEvent event) {
                                 }
                                 break;
                             case "error":
-                                throw new StreamErrorException(PacketParserUtils.parseStreamError(parser));
+                                //Some bosh error isn't stream error.
```

Comment:
That appears to be a sensible change. Could you rebase it in a single commit (and ideally issue an extra PR for it)? Looks like something we want to have in the 4.2 branch, for the upcoming 4.2.0 release in 2-3 weeks.